ElectronClient/gui/ConfigScreen.jsx

```diff
@@ -270,8 +280,22 @@ class ConfigScreenComponent extends React.Component {
 			borderRadius: 4,
 		});
 
+		const detectTheme = (value) => {
+			this.setState({ shouldUseDarkColors: bridge().shouldUseDarkColors() });
+
+			// active theme should change to preferred theme based on system detected theme
+			if (value == true) {
+				this.state.shouldUseDarkColors ? updateSettingValue('theme', settings['preferredDarkTheme']) : updateSettingValue('theme', settings['preferredLightTheme']);
+			} else {
+				updateSettingValue('theme', settings['theme']);
```

Comment:
Can you please help me understand how does this line works?  It looks like it is setting the theme to its current value.

Reply:
This `detectTheme` function called every time when we check/uncheck Auto Detect option so when we check the Auto Detect option the value will be true now it means we want to set current theme to System Detected theme which can be dark/light based on the system theme and here 
```javascript
this.state.shouldUseDarkColors ? updateSettingValue('theme', settings['preferredDarkTheme']) : updateSettingValue('theme', settings['preferredLightTheme']);
```
we are checking that if system theme is DARK then the `theme` should be set to whatever theme is selected in `preferredDarkTheme` dropdown and similarly if system theme is LIGHT then the `theme` should be set to whatever theme is selected in `preferredLightTheme` dropdown and finally when we click on `Apply` button the changes will take affect. (As shown in GIF in the first comment)

`updateSettingValue('theme', settings['theme']);`
Now consider the case when we uncheck the Auto Detect option then obviously value will be false in that case `theme` should set to whatever theme is selected in `Theme` dropdown and  `preferredDarkTheme` and `preferredLightTheme` settings shouldn't take affect they should only works when `Auto Detect` option is checked. 

Generally, `preferredDarkTheme` and `preferredLightTheme` will always work for `Night Mode` button.